sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/ProcessEventArgs.cs

```diff
@@ -14,20 +15,26 @@ namespace Azure.Messaging.EventHubs.Processor
     ///   also provides a way of creating a checkpoint based on the information contained in the associated event.
     /// </summary>
     ///
-    public struct EventProcessorEvent
+    public struct ProcessEventArgs
     {
         /// <summary>
         ///   The context of the Event Hub partition this instance is associated with.
         /// </summary>
         ///
-        public PartitionContext Context { get; }
+        public PartitionContext Partition { get; }
 
         /// <summary>
         ///   The received event to be processed.  Expected to be <c>null</c> if the receive call has timed out.
         /// </summary>
         ///
         public EventData Data { get; }
```

Comment:
In the previous library, events where processed in batch, and that seems like a more efficient way of doing things (for example, batch saving them to a file is a lot faster than one-by-one).
What's the rationale behind switching to one-by-one?

Reply:
The intent is to better match the pattern used by the iterators, where you're pulling events one-by-one to ease the "step-up" experience that we were discussing.  Behind the scenes, events are read eagerly into a bounded channel from the service and not one-by-one, so there's no loss of efficiency from a service interaction standpoint.   The question of whether it is better, faster, and easier to process in a group or by single will vary between scenarios and developer preferences - here, again, there's no one-size-fits-all approach that would be clearly better.

For those who prefer to perform downstream processing in batches, it's easy enough to do the grouping in the handler or by overriding the `OnProcessEvent` method to collect the events and then ti dispatch them as a batch.  We're planning to provide samples that illustrate the approaches as it is recognized as a common scenario.

Reply:
I did think about the user doing its own batching, but I'm not sure how easy it will be with regards to checkpoints...

Reply:
Checkpointing was always something that was under developer control, where creation of a checkpoint was concerned.  There are different patterns and strategies in use today which vary by individual developer and scenario   Some scenarios call for more frequent checkpoints because events are expensive to process and the performance hit of writing the checkpoint is acceptable.  On the other extreme, some events are cheap to process and maintaining a high throughput makes it preferable to reprocess events rather than take the performance hit of writing out a checkpoint.  Still other scenarios used a strict time-based checkpointing approach where a checkpoint was taken on a fixed interval no matter how many events had been processed.    

In the current model, each event emitted to the processing delegate can be used to create a checkpoint by calling the method on its context.    This can be used in whatever pattern makes sense in a given scenario.   In the batching scenario, for example, should you wish to checkpoint once-per-batch, you would call the method on the last event received before dispatching the batch for processing.